woocommerce/woocommerce-android
# Free Trial: Add free trial store creation flow tracks

Summary
==========
Android counterpart of https://github.com/woocommerce/woocommerce-ios/pull/9336

Fix issue #8495 by adding a `isFreeTrial` parameter to the following four store creation track events:

- `login_prologue_create_site_tapped`
- `login_woocommerce_site_created`
-  `site_creation_failed`
-  `site_creation_dismissed`

The idea is to easily filter the paid store creation flow from the free trial one when looping back to see the store creation analytics.

With the changes introduced here, the isFreeTrial will be true when the Free Trial Feature Flag is active and uses the native creation flow, but when we remove the flag, we should replace the parameter with always true for all the native flow. In case we come back to providing two different store creation flows back again, then this parameter shall be set depending on the creation type.

How to Test
==========
1. Create a Free Trial store and verify if the Track events keep exactly the same as before, but the described four track events contain the `isFreeTrial` parameter value as `true`.

Update release notes:
